crates/toolchain/instructions/src/program.rs

```diff
@@ -58,6 +58,27 @@ impl<F: Field> Program<F> {
         }
     }
 
+    pub fn new_without_debug_infos_with_option(
+        instructions: &[Option<Instruction<F>>],
```

Comment:
I wonder if we should change `Program` to have parallel `instructions` and `debug_info` rather than a vec of pairs?

Reply:
hm that could be good - `debug_info` is weird and only for dsl